pgmpy/base/DAG.py

```diff
@@ -1828,6 +1829,100 @@ def edge_strength(self, data, edges=None):
+            Dataset to be used to run the scoring methods/tests
+
+        metrics: tuple (Callable or strings)
+            A list of the metrics that are to be run on the model and data
+
+        significance_level: float
+            To compare with p-value to count the failing CIs (conditional independencies). A
+            default value of 0.05 is provided which can be overwriteen by user based on need
+
+        **kwargs:
+            Any parameter that needs to be customized in the call to different metric based methods.
+            This includes all parameters that affect the metric calls in pgmpy.metrics.metrics.
+
+        Returns
+        ----------
+        df_result: pandas.Dataframe
+            A dataframe of all the metric results run on the given causal model
+            using the dataset provided.
+        """
+        # checking if data provided is valid
+        if not isinstance(data, pd.DataFrame) or data is None:
+            raise ValueError(
+                f"data must be a pandas.DataFrame instance. Got {type(data)}"
+            )
+
+        from pgmpy.estimators.CITests import get_callable_ci_test
+        from pgmpy.metrics.metrics import get_metrics
+        from pgmpy.utils import get_dataset_type
+
+        # List of method calls generated according to the metric list provided
+        callable_metrics = get_metrics(metrics=metrics)
+
+        # Getting the CI Test method
+        kwargs["ci_test"] = get_callable_ci_test(test=None, data=data)
+
+        # To store final results of the tests
+        metric_vals = {}
+
+        for n, c in callable_metrics.items():
+            sig = inspect.signature(c)
+            valid_params = sig.parameters.keys()
+            filtered_kwargs = {k: v for k, v in kwargs.items() if k in valid_params}
+
+            if n in ["aic", "bic"]:
+                var_type = get_dataset_type(data)
+                if var_type == "continuous":
+                    suffix = "g"
+                elif var_type == "discrete":
+                    suffix = "d"
+                else:
+                    suffix = "cg"
+                filtered_kwargs["scoring_method"] = f"{n}-" + suffix
+
+            result = c(model=self, data=data, **filtered_kwargs)
+
+            if n in ["correlation", "log-likelihood", "aic", "bic"]:
+                metric_vals[n.capitalize()] = result
+            elif n == "fisher-c":
+                if isinstance(result, tuple):
+                    (metric_vals["Fisher-C p-value"], metric_vals["RMSEA"]) = result
+                else:
+                    metric_vals["Fisher-C p-value"] = result
+            elif n == "implied-cis":
+                metric_vals["Failing CIs / Total CIs"] = (
+                    f"{(result["p-value"] < significance_level).sum()} / {len(result)}"
+                )
+
+        df_result = pd.DataFrame(
+            {"METRIC": list(metric_vals.keys()), "RESULT": list(metric_vals.values())}
+        )
```

Comment:
I would suggest making the metric names as the index of the dataframe rather than being a column.